src/test/unit/specific_feature_tests/dialogs.js

```diff
@@ -41,48 +41,27 @@ module("dialogs-opening_or_not", {setup: prepareUnitTestModule});
  */
 /* jshint latedef: nofunc */
 function dialogTestHelper(args) {
-    // Sinon doesn't allow wrapping of `window.open` with a spy because `open`
-    // doesn't look like a function in these browsers.
-    var sinonCantWrapWindowOpen = jQuery.browser.msie &&
-        jQuery.browser.versionNumber <= 8;
+    var wymeditor = jQuery.wymeditors(0);
+
+    // Wrap the dialog function with a spy
+    args.currentTest.spy(wymeditor, "dialog");
 
     manipulationTestHelper({
         startHtml: args.noChangeHtml,
         setCaretInSelector: args.setCaretInSelector,
-        prepareFunc: prepareFunc,
+        prepareFunc: args.prepareFunc,
         manipulationClickSelector:
             getDialogToolbarSelector(args.dialogName),
+        manipulationKeyCombo:
+            getDialogKeyCombo(args.dialogName),
         expectedResultHtml: args.noChangeHtml,
         additionalAssertionsFunc: additionalAssertionsFunc
     });
 
-    function prepareFunc(wymeditor) {
-        // Wrap the dialog function with a spy
-        args.currentTest.spy(wymeditor, "dialog");
-        if (sinonCantWrapWindowOpen !== true) {
-            // Wrap the `window.open` native function with a spy
-            args.currentTest.spy(window, "open");
-        }
-
-        if (typeof args.prepareFunc === "function") {
-            // Call the provided `prepareFunc`
-            args.prepareFunc(wymeditor);
-        }
-    }
-
     function additionalAssertionsFunc(wymeditor) {
-        var dialogWindowOrFalse;
-        expect(expect() + 1);
-
-        ok(
-            // The dialog function is called, whether the dialog is to be
-            // opened or not. It has the code that determines that.
-            wymeditor.dialog.calledOnce,
-            "Dialog function called once"
-        );
-
         // Its return value is either the dialog window or `false`.
-        dialogWindowOrFalse = wymeditor.dialog.returnValues[0];
+        var dialogReturns = wymeditor.dialog.returnValues,
+            dialogWindowOrFalse = dialogReturns[dialogReturns.length - 1];
```

Comment:
`additionalAssertionsFunc` for the dialogs tests is to get the return value of the wrapped `dialog` method, which is either the dialog window or `false`.

The change here is that now we have multiple return values, because one of the dialogs (link) opens using a key combo, as well.

So for each `manipulationCause` we get the return value (the last array member).